scene/resources/particle_process_material.cpp

```diff
@@ -984,14 +980,18 @@ void ParticleProcessMaterial::_update_shader() {
 		}
 		code += "		\n";
 		code += "		vec3 noise_direction = get_noise_direction(TRANSFORM[3].xyz);\n";
-		// The following snippet causes massive performance hit. We don't need it as long as collision is disabled.
-		// Refer to GH-83744 for more info.
-		if (collision_mode != COLLISION_DISABLED) {
+
+		// Godot detects when the COLLIDED keyword is used. If it's used anywhere in the shader then Godot will generate the screen space SDF for collisions.
+		// We don't need it as long as collision is disabled. Refer to GH-83744 for more info.
+		if (collision_mode == COLLISION_RIGID) {
 			code += "		if (!COLLIDED) {\n";
-			code += "			\n";
+		}
+		if (collision_mode != COLLISION_DISABLED) {
```

Comment:
I don't think this should have been added. This code should _always_ run when collisions are disabled. 

If collisions are set to rigid mode, then this code should only run when ``!COLLIDED`` is true.

Reply:
Sorry for the misunderstanding, I was a bit lost.